### PR TITLE
Memory Pressure demo

### PR DESCRIPTION
**Pull request by orviwan**

### CODE REVIEW

File: src/rocky/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+rocky.on('memorypressure', function(event) {
+  switch(event.level) {
+    case 'high':
+      // Memory pressure is high, free some memory
+      dataLogs = [];
+      console.log('relieved the pressure');
+      break;
+    case 'normal':
+    case 'low':
+      // Not implemented
```
Comment (HBehrens) [inline src/rocky/index.js]: could you still log this? maybe as the first line of the handler? `console.log('memory pressure event', event.level);` or something?

File: src/rocky/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      break;
+  }
+});
+
+rocky.on('minutechange', function(event) {
+  rocky.requestDraw();
+  // Fill the logs to increase memory pressure
```
Comment (HBehrens) [inline src/rocky/index.js]: the term "log" is a bit misleading here.

File: src/rocky/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      console.log('relieved the pressure');
+      break;
+    case 'normal':
+    case 'low':
+      // Not implemented
+      break;
+  }
+});
+
+rocky.on('minutechange', function(event) {
```
Comment (HBehrens) [inline src/rocky/index.js]: if the purpose of this test is to regularly hit pressure, doesn't it make more sense to trigger the overflow more often, e.g. `secondchange` and then something line `seconds % 5 == 0` ?

File: src/rocky/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+// Draw the current time on screen
+rocky.on('draw', function(event) {
+  var ctx = event.context;
+  ctx.clearRect(0, 0, ctx.canvas.clientWidth, ctx.canvas.clientHeight);
+  var w = ctx.canvas.unobstructedWidth;
+  var h = ctx.canvas.unobstructedHeight;
+  ctx.fillStyle = 'white';
+  ctx.textAlign = 'center';
+  ctx.fillText(new Date().toLocaleTimeString(), w / 2, h / 2, w);
```
Comment (HBehrens) [inline src/rocky/index.js]: does it make sense to print how often we hit the event? you could increase a global counter and show it here.

**Review by HBehrens — COMMENTED** [review]
I'd emphasize a bit more on the effect. See my inline comments about this.

**Comment by orviwan** [comment]
@HBehrens ready for another review!

File: src/rocky/index.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+var rocky = require('rocky');
+
+var dataLogs = [];
+var memoryPressureHitCount = 0;
+
+// Triggered when memory pressure is an issue
+rocky.on('memorypressure', function(event) {
+  console.log('memory pressure event: ' + event.level);
```
Comment (martijnthe) [inline src/rocky/index.js]: Probably best to log this after releasing the pressure, just to be sure that there's enough memory to run this `console.log(...)` line. Perhaps even add a comment about this?

**Comment by martijnthe** [comment]
Thanks! LGTM

**Comment by mjessome** [comment]
:+1:

**Pull request merged**